ui/app/routes/vault/cluster/secrets/backend/configuration/index.js

```diff
@@ -73,112 +37,121 @@ export default class SecretsBackendConfigurationRoute extends Route {
       }
-      const configArray = [];
-      if (azureModel.isConfigured) configArray.push(azureModel);
-      if (issuer) configArray.push(issuer);
-      return configArray;
     } catch (e) {
-      if (e.httpStatus === 404) {
+      const error = await this.parseApiError(e);
+      if (error.httpStatus === 404) {
         // a 404 error is thrown when Azure's config hasn't been set yet.
         return;
       }
-      throw e;
+      throw error;
     }
   }
 
-  async fetchGcpConfig(id) {
+  async fetchGcpConfig(path) {
     try {
-      const gcpModel = await this.store.queryRecord('gcp/config', { backend: id });
-      let issuer = null;
-      if (this.version.isEnterprise) {
-        const WIF_FIELDS = ['identityTokenAudience', 'identityTokenTtl', 'serviceAccountEmail'];
-        WIF_FIELDS.some((field) => gcpModel[field]) ? (issuer = await this.fetchIssuer()) : null;
+      const { data: gcpConfig } = await this.api.secrets.googleCloudReadConfiguration(path);
+      const WIF_FIELDS = ['identityTokenAudience', 'identityTokenTtl', 'serviceAccountEmail'];
+      const issuer = await this.checkIssuer(gcpConfig, WIF_FIELDS);
+
+      if (gcpConfig) {
+        return Object.assign(gcpConfig, issuer);
       }
-      const configArray = [];
-      if (gcpModel) configArray.push(gcpModel);
-      if (issuer) configArray.push(issuer);
-      return configArray;
     } catch (e) {
-      if (e.httpStatus === 404) {
+      const error = await this.parseApiError(e);
+      if (error.httpStatus === 404) {
         // a 404 error is thrown when GCP's config hasn't been set yet.
         return;
       }
-      throw e;
+      throw error;
     }
   }
 
-  async fetchIssuer() {
+  async fetchSshCaConfig(path) {
     try {
-      return await this.store.queryRecord('identity/oidc/config', {});
+      const { data } = await this.api.secrets.sshReadCaConfiguration(path);
+      return data;
     } catch (e) {
-      // silently fail if the endpoint is not available or the user doesn't have permission to access it.
-      return;
-    }
-  }
-
-  async fetchSshCaConfig(id) {
-    try {
-      return await this.store.queryRecord('ssh/ca-config', { backend: id });
-    } catch (e) {
-      if (e.httpStatus === 400 && e.errors[0] === `keys haven't been configured yet`) {
+      const error = await this.parseApiError(e);
+      if (error.httpStatus === 400 && error.errors[0] === `keys haven't been configured yet`) {
         // When first mounting a SSH engine it throws a 400 error with this specific message.
         // We want to catch this situation and return nothing so that the component can handle it correctly.
         return;
       }
-      throw e;
+      throw error;
     }
   }
 
+  async checkIssuer(config, fields) {
+    // issuer is an enterprise only related feature
+    // issuer is also a global endpoint that doesn't mean anything in the AWS secret details context if WIF related fields on the rootConfig have not been set.
```

Comment:
When I wrote this comment it was AWS specific as it was within the aws method, but here it is not aws specific
```suggestion
    // issuer is also a global endpoint that doesn't mean anything in the secret details context if WIF related fields on the engine's config have not been set.
```

Reply:
Good catch thanks!